api/render/render.go

```diff
@@ -24,14 +23,9 @@ func JSON(w http.ResponseWriter, v interface{}) {
 // JSONStatus sets the Content-Type of w to application/json unless one is
 // specified.
 func JSONStatus(w http.ResponseWriter, v interface{}, status int) {
-	var b bytes.Buffer
-	if err := json.NewEncoder(&b).Encode(v); err != nil {
-		panic(err)
-	}
-
 	setContentTypeUnlessPresent(w, "application/json")
 	w.WriteHeader(status)
-	_, _ = b.WriteTo(w)
+	_ = json.NewEncoder(w).Encode(v)
```

Comment:
You could panic or log here if an error was returned?

Reply:
I've pushed a change that maintains the old behavior (i.e. panics) in case of a `json.UnsupportedValueError` and `json.UnsupportedTypeError` errors.

I think this last version is the one we should go with.

Reply:
The reason I want to keep the panic is that even if something's written in the log the response will still arrive as 200 to the caller.

If either of these error types bubble up, I think, it's an indicator that something unintended has happened in our code (like a channel being marshaled) so I'd like these cases to remain as panics.